lib/datadog/tracing/contrib/sidekiq/distributed/propagation.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+# typed: true
+
+require_relative 'fetcher'
+require_relative '../../../distributed/propagation'
+require_relative '../../../distributed/b3_multi'
+require_relative '../../../distributed/b3_single'
+require_relative '../../../distributed/datadog'
+require_relative '../../../distributed/none'
+require_relative '../../../distributed/trace_context'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Sidekiq
+        module Distributed
+          # Extracts and injects propagation through HTTP headers.
+          class Propagation < Tracing::Distributed::Propagation
+            def initialize
+              super(
+                propagation_styles: {
+                  Tracing::Configuration::Ext::Distributed::PROPAGATION_STYLE_B3_MULTI_HEADER =>
+                    Tracing::Distributed::B3Multi.new(fetcher: Fetcher),
```

Comment:
You can use `Tracing::Distributed::Fetcher` directly here, so you can delete the file `lib/datadog/tracing/contrib/sidekiq/distributed/fetcher.rb`:
```suggestion
                    Tracing::Distributed::B3Multi.new(fetcher: Tracing::Distributed::Fetcher),
```